packages/svelte/src/internal/client/reactivity/store.js

```diff
@@ -18,30 +18,25 @@ import { mutable_source, set } from './sources.js';
  * @returns {V}
  */
 export function store_get(store, store_name, stores) {
-	/** @type {StoreReferencesContainer[''] | undefined} */
-	let entry = stores[store_name];
-	const is_new = entry === undefined;
-
-	if (is_new) {
-		entry = {
-			store: null,
-			last_value: null,
-			value: mutable_source(UNINITIALIZED),
-			unsubscribe: noop
-		};
-		stores[store_name] = entry;
-	}
+	const entry = (stores[store_name] ??= {
+		store: null,
+		source: mutable_source(UNINITIALIZED),
+		unsubscribe: noop
+	});
 
-	if (is_new || entry.store !== store) {
+	if (entry.store !== store) {
 		entry.unsubscribe();
 		entry.store = store ?? null;
-		entry.unsubscribe = connect_store_to_signal(store, entry.value);
+
+		if (store == null) {
+			set(entry.source, undefined);
+			entry.unsubscribe = noop;
+		} else {
+			entry.unsubscribe = subscribe_to_store(store, (v) => set(entry.source, v));
+		}
 	}
 
-	const value = get(entry.value);
-	// This could happen if the store was cleaned up because the component was destroyed and there's a leak on the user side.
-	// In that case we don't want to fail with a cryptic Symbol error, but rather return the last value we got.
-	return value === UNINITIALIZED ? entry.last_value : value;
```

Comment:
This was in response to a real bug I investigated at some point. I think the fix was gradually removed over time (there was more logic there before) so that it's unused now. Let me see if I can dig up the PR and related issue to see if something like this is still necessary.

Reply:
It was from https://github.com/sveltejs/svelte-octane/commit/6ebe7118c8fe3f45d0b9787d1e9c92cc7e27d62d, but without any reference - I vaguely remember that it occured while testing the code out on some Svelte site and noticing the bug. But this error can't happen anymore now because back then the source value was set back to `UNINITIALIZED` which is no longer the case now - so all good.